src/perspective.jl

```diff
@@ -0,0 +1,95 @@
+"""
+    PerspectiveMap()
+
+Construct a perspective transformation. The persepective transformation takes,
+e.g., a point in 3D space and "projects" it onto a 2D virtual screen of an ideal
+pinhole camera (at distance `1` away from the camera). The camera is oriented
+towards the positive-Z axis (or in general, along the final dimension) and the
+sign of the `x` and `y` components is preserved for objects in front of the
+camera (objects behind the camera are also projected and therefore inverted - it
+is up to the user to cull these as necessary).
+
+This transformation is designed to be used in composition with other coordinate
+transformations, defining e.g. the position and orientation of the camera. For
+example:
+
+    cam_transform = PerspectiveMap() ∘ inv(AffineMap(cam_rotation, cam_position))
+    screen_points = map(cam_transform, points)
+
+(see also `cameramap`)
+"""
+immutable PerspectiveMap <: Transformation
+end
+
+function (::PerspectiveMap)(v::AbstractVector)
+    scale =  1/v[end]
+    return [v[i] * scale for i in 1:length(v)-1]
+end
+
+@inline function (::PerspectiveMap)(v::StaticVector)
+    return pop(v) * inv(v[end])
+end
+
+Base.@pure Base.isapprox(::PerspectiveMap, ::PerspectiveMap; kwargs...) = true
+
+"""
+    cameramap(property = value, ...)
+
+Create a transformation that takes points in real space (e.g. 3D) and projects
+them through a perspective transformation onto the focal plane of an ideal
+(pinhole) camera with the given properties.
+
+All properties are optional. Valid properties include:
+
+ * `focal_length` (in physical units)
+ * `pixel_size` (in physical units) or `pixel_size_x` and `pixel_size_y`
+ * `offset_x` and `offset_y` (in pixels)
+ * `origin` (a vector) and `orientation` (a rotation matrix)
+
+By default, the camera looks towards the postive-`z` axis from `(0,0,0)` and
+the sign of the `x` and `y` components is preserved for objects in front of the
+camera (objects behind the camera are also projected and therefor inverted - it
+is up to the user to cull these as necessary).
+
+If `origin` and `orientation` are specified, the camera is translated to `origin`
+and rotated by `orientation` before the perspective map is applied.
+
+(see also `PerspectiveMap`)
+"""
+function cameramap(;focal_length = nothing,
```

Comment:
I like this function as an example of how the perspective transformation can be used. However I'm not so sure that people will really want pixel (raster) coordinates in general. Sometimes they'll want a lens distortion interspersed, sometimes they'll want to stop at normalized texture coordinates, sometimes a normalized coordinate system where the optical axis is at the origin, etc.

So I think it's quite unlikely that we can satisfy people with a single function, and I don't think this actually covers what we need in practise. @PaulBellette @awbsmith what do you think? Perhaps clear docs about how to construct a few definitive transformations would be best.